### ISSUE TITLE
Tabular Learning in mlr3torch

### ISSUE THREAD

**Issue by pfistfl** [mlr-org/mlr3torch#25]
Summary

On a meta-level we might first want to support classification and then extend to e.g. regression. 
This requires that functionality is provided through functions which can be called for all task types. 

In Scope: 
- [x] Build a dataloader / dataset from tabular data.
  - [x] Takes care of converting labels to int
  - [x] Takes care of converting categoricals to int / tuples etc. 

- [ ] High-Level API: Pre-defined NN architectures for tabular data 
   - [x] TabNet (tabnet package)
     - [ ] See also #13  
   - [ ] Deep & Wide (mlr3keras)
   - [x] FCN (see Pytorch Tabular)
   - [x] Tabular ResNet (paper)
   - [ ] See #14
 - [ ] Vignette with comparison of the learners. 
- [ ] Custom API:
  - [ ] User provides a `nn.Module`
  - [ ] Good documentation with an example.
- [ ] Low-Level API:
  - [ ] `top()`: Build a graph from torch operations
  - [ ] A set of building blocks to build up graphs using `top()`:
    - [ ] Embeddings
    - [ ] Fully Connected
    - [ ] Residual Blocks
  - [ ] Vignette on how to use it with few interesting examples
